src/core_plugins/timelion/server/lib/offset_time.js

```diff
@@ -34,3 +34,29 @@ export default function offsetTime(milliseconds, offset, reverse) {
   const momentObj = moment(milliseconds)[mode](parts[1], parts[2]);
   return momentObj.valueOf();
 }
+
+/*
+ * Calculate offset when parameter is requesting a relative offset based on requested time range.
+ *
+ * @param {string} offset - offset parameter value
+ * @param {number} from - kibana global time 'from' in milliseconds
+ * @param {number} to - kibana global time 'to' in milliseconds
+ */
+export function preprocessOffset(offset, from, to) {
+  if (!offset.startsWith('kibana_time')) {
```

Comment:
I would rather not want to name this `kibana_time`. We can technically already have different timeranges per dashboard panel and will also expose that in the future via a UI to the user, also when we are moving to the Canvas pipeline for rendering all visualizations, that timerange that makes up the visualization could potentially come from anywhere (any other function in the pipeline, that provides it).

Thus I would rather vote for using a string like `timerange`.

I would also change in that case the above help text from:

`Offset the series relative to Kibana\'s time range by using the value "kibana_time". Eg kibana_time:-2 will specify an offset that is twice the selected kibana time.` to
`Offset the series relative to the charts overall time range, by using the value "timerange", e.g. "timerange:-2" will specify an offset that is twice the overall chart time range to the past.`

Reply:
++